controller/telemetry/server.go

```diff
@@ -264,9 +274,38 @@ func (s *server) Report(ctx context.Context, req *write.ReportRequest) (*write.R
 		id = req.Process.ScheduledNamespace + "/" + req.Process.ScheduledInstance
 	}
 
-	log := log.WithFields(log.Fields{"id": id})
-	log.Debugf("received report with %d requests", len(req.Requests))
+	logCtx := log.WithFields(log.Fields{"id": id})
+	logCtx.Debugf("received report with %d requests", len(req.Requests))
+
+	// validate the request's latency histogram.
+	numBuckets := len(req.LatencyMaxValues)
+	// add one for the +Inf bucket
+	expectedNumBuckets := len(responseLatencyBuckets) + 1
+	if numBuckets != expectedNumBuckets {
+		err := errors.New(
+			"received report with incorrect number of latency buckets")
+		logCtx.WithFields(log.Fields{
+			"numBuckets": numBuckets,
+			"expected":   expectedNumBuckets,
+		}).WithError(err).Error()
+		return nil, err
+	}
 
+	for bucketNumber, bucketMax := range req.LatencyMaxValues {
```

Comment:
I don't think we want this check.  It should be possible for the proxy to pass a differently bucketed histogram to the controller (for example if we change the buckets in a future version and upgrade the proxy before upgrading the controller).